### PR TITLE
General update deployment names

### LINKED ISSUE #209: General - Unique deployment names (from multi-region testing)
**IMPORTANT suggest tagging the deployment name with timestamp/hash + the cloudmarque version (Get-CmAzContext.PackageVersion)**

Add unique timestamp to deployment names to preserve deployment history / traceability and avoid this type of error. Test in two regions to validate fixe works.

Also see New-CmAzIaasNetworking - deployments not timestamp #199 but relates to all commandlets

InvalidDeploymentLocation - Long running operation failed with status 'Failed'. Additional Info:'Invalid deployment location 'UK West'. The deployment
| 'Cm_network_resource_group_deployment' already exists in location 'uksouth'.'

### PR DESCRIPTION
**Pull request by deepend-dev**
closes #209

### CODE REVIEW

File: Cloudmarque.Azure/Public/Deploy/Core/budgets/New-CmAzCoreBillingRule.ps1 [hunk 1]
```diff
@@ -90,7 +90,11 @@ function New-CmAzCoreBillingRule {
 			}
 
 			Write-Verbose "Deploying budgets..."
+
+			$deploymentName = Get-CmAzResourceName -Resource "Deployment" -Architecture "Core" -Region $SettingsObject.location -Name "New-CmAzCoreBillingRule"
+
 			New-AzDeployment `
+				-Name $deploymentName `
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/Core/budgets/New-CmAzCoreBillingRule.ps1:97]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/Core/keyvault/New-CmAzCoreKeyVault.ps1 [hunk 2]
```diff
@@ -98,7 +98,11 @@ function New-CmAzCoreKeyVault {
 			}
 
 			Write-Verbose "Deploying Keyvaults..."
+
+			$deploymentName = Get-CmAzResourceName -Resource "Deployment" -Architecture "Core" -Region $SettingsObject.location -Name "New-CmAzCoreKeyVault"
+
 			New-AzResourceGroupDeployment `
+				-Name $deploymentName `
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/Core/keyvault/New-CmAzCoreKeyVault.ps1:105]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/Core/monitoring/New-CmAzCoreMonitor.ps1 [hunk 3]
```diff
@@ -197,7 +197,11 @@ function New-CmAzCoreMonitor {
 			New-CmAzIaasStorage -SettingsObject $storageObject -OmitTags
 
 			Write-Verbose "Deploying monitor resources..."
+
+			$deploymentName = Get-CmAzResourceName -Resource "Deployment" -Architecture "Core" -Region $SettingsObject.Location -Name "New-CmAzCoreMonitor"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/Core/monitoring/New-CmAzCoreMonitor.ps1:201]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/bastionhost/New-CmAzIaasBastionHost.ps1 [hunk 4]
```diff
@@ -92,7 +92,13 @@
 				Set-GlobalServiceValues -GlobalServiceContainer $SettingsObject -ServiceKey "bastionPublicIP" -ResourceServiceContainer $bastionHost
 			}
 
-			New-AzResourceGroupDeployment -ResourceGroupName $resourceGroupName `
+			Write-Verbose "Deploying Bastion Hosts.."
+
+			$deploymentName = Get-CmAzResourceName 	-Resource "Deployment" -Architecture "IaaS" -Region $SettingsObject.location -Name "New-CmAzIaasBastionHost"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/bastionhost/New-CmAzIaasBastionHost.ps1:97]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/networking/New-CmAzIaasNetworking.ps1 [hunk 5]
```diff
@@ -878,8 +878,11 @@
 
 			# Arm Deployment
 			Write-Verbose "Deploying resource groups..."
+
+			$deploymentNameRg = Get-CmAzResourceName -Resource "Deployment" -Architecture "IaaS" -Region $resourceGroupObjectArray[0].resourceGroup.location -Name "New-CmAzIaasNetworking-RGs"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/networking/New-CmAzIaasNetworking.ps1:882]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/networking/New-CmAzIaasNetworking.ps1 [hunk 6]
```diff
@@ -878,8 +878,11 @@
...
+
 			New-AzDeployment `
-				-Name "Cm_network_resource_group_deployment_$(Get-Date -Format 'ddMMyyyyHHmmsssss')" `
+				-Name $deploymentNameRg `
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/networking/New-CmAzIaasNetworking.ps1:885]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/networking/New-CmAzIaasNetworking.ps1 [hunk 7]
```diff
@@ -947,7 +950,11 @@
 				New-AzResourceGroup -Location $workspace.location -Name $networkWatcherResourceGroupName -Force
 
 				Write-Verbose "Deploying nsgs..."
+
+				$deploymentNameNsg = Get-CmAzResourceName -Resource "Deployment" -Architecture "IaaS" -Region $workspace.location -Name "New-CmAzIaasNetworking-Nsgs"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/networking/New-CmAzIaasNetworking.ps1:954]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/networking/New-CmAzIaasNetworking.ps1 [hunk 8]
```diff
@@ -957,7 +964,11 @@
 			}
 
 			Write-Verbose "Deploying vnets and udrs..."
+
+			$deploymentNameVu = Get-CmAzResourceName -Resource "Deployment" -Architecture "IaaS" -Region $resourceGroupObjectArray[0].resourceGroup.location -Name "New-CmAzIaasNetworking-VUs"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/networking/New-CmAzIaasNetworking.ps1:968]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/networking/New-CmAzIaasNetworking.ps1 [hunk 9]
```diff
@@ -987,7 +998,11 @@
 				}
 
 				Write-Verbose "Configuring vnet peerings..."
+
+				$deploymentNamePeerings = Get-CmAzResourceName -Resource "Deployment" -Architecture "IaaS" -Region $resourceGroupObjectArray[0].resourceGroup.location -Name "New-CmAzIaasNetworking-Vps"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/networking/New-CmAzIaasNetworking.ps1:1002]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/recoveryvault/New-CmAzIaasRecoveryServicesVault.ps1 [hunk 10]
```diff
@@ -84,24 +84,32 @@ function New-CmAzIaasRecoveryServicesVault {
...
 
             Set-GlobalServiceValues -GlobalServiceContainer $SettingsObject -ServiceKey "recoveryVault" -ResourceServiceContainer $vault
         }
 
         $workspace = Get-CmAzService -Service $SettingsObject.service.dependencies.workspace -Region $SettingsObject.location -ThrowIfUnavailable -ThrowIfMultiple
 
-        Write-Verbose "Deploying Azure ARM template for Recovery Services Vaults."
+        Write-Verbose "Deploying Azure ARM template for Recovery Services Vaults..."
+
+        $deploymentNameRv = Get-CmAzResourceName -Resource "Deployment" -Architecture "IaaS" -Region $SettingsObject.location -Name "New-CmAzIaasRecoveryServicesVault"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/recoveryvault/New-CmAzIaasRecoveryServicesVault.ps1:97]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/recoveryvault/New-CmAzIaasRecoveryServicesVault.ps1 [hunk 11]
```diff
@@ -84,24 +84,32 @@ function New-CmAzIaasRecoveryServicesVault {
...
             -ResourceGroupName $resourceGroupName `
             -TemplateFile "$PSScriptRoot\New-CmAzIaasRecoveryServicesVault.json" `
             -RecoveryServicesVaults $SettingsObject.recoveryServicesVaults `
             -Force
 
-        Write-Verbose "Deploying Azure ARM template for Recovery Services Diagnostics."
+        Write-Verbose "Deploying Recovery Services Diagnostics..."
         Foreach ($vault in $SettingsObject.recoveryServicesVaults) {
+
+            $deploymentNameRvSd = Get-CmAzResourceName -Resource "Deployment" -Architecture "IaaS" -Region $SettingsObject.location -Name "New-CmAzIaasRecoveryServicesVault-Sd"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/recoveryvault/New-CmAzIaasRecoveryServicesVault.ps1:109]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/recoveryvault/New-CmAzIaasRecoveryServicesVault.ps1 [hunk 12]
```diff
@@ -111,9 +119,13 @@ function New-CmAzIaasRecoveryServicesVault {
 
         if ($PolicySettingsObject) {
 
-            Write-Verbose "Deploying Azure ARM template for Recovery Services Policies."
+            Write-Verbose "Deploying Recovery Services Policies..."
             Foreach ($vault in $SettingsObject.recoveryServicesVaults) {
+
+                $deploymentNameRvSp = Get-CmAzResourceName -Resource "Deployment" -Architecture "IaaS" -Region $SettingsObject.location -Name "New-CmAzIaasRecoveryServicesVault-Sp"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/recoveryvault/New-CmAzIaasRecoveryServicesVault.ps1:125]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/storage/New-CmAzIaasStorage.ps1 [hunk 13]
```diff
@@ -181,8 +181,12 @@
 				Write-Error "Please provide appropriate service tag for existing resource group or provide unique name to create new."
 			}
 
+			Write-Verbose "Deploying Storage Accounts..."
+
+			$deploymentName = Get-CmAzResourceName -Resource "Deployment" -Region $SettingsObject.Location -Architecture "IaaS" -Name "New-CmAzIaasStorage"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/storage/New-CmAzIaasStorage.ps1:186]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/vm/New-CmAzIaasVm.ps1 [hunk 14]
```diff
@@ -206,7 +206,11 @@
 				Write-Verbose "No valid virtual machines available for deployment..."
 			}
 			else {
+
+				$deploymentNameRgs = Get-CmAzResourceName -Resource "Deployment" -Architecture "IaaS" -Region $SettingsObject.location -Name "New-CmAzIaasVm-Rgs"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/vm/New-CmAzIaasVm.ps1:210]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/vm/New-CmAzIaasVm.ps1 [hunk 15]
```diff
@@ -220,7 +224,10 @@
 					"LocalAdminPassword" = ConvertFrom-SecureString $LocalAdminPassword -AsPlainText;
 				}
 
+				$deploymentNameVm = Get-CmAzResourceName -Resource "Deployment" -Architecture "IaaS" -Region $SettingsObject.location -Name "New-CmAzIaasVm"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/vm/New-CmAzIaasVm.ps1:227]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/vm/Set-CmAzIaasUpdateManagement.ps1 [hunk 16]
```diff
@@ -14,13 +14,13 @@ function Set-CmAzIaasUpdateManagement {
 		 Object containing the configuration values required to run this cmdlet.
 
 		.Component
-		 Core
+		 IaaS
 
 		.Example
-		 Set-CmAzCoreMonitorLoggingAutomation -SettingsFile "c:/directory/settingsFile.yml"
+		 Set-CmAzIaaSMonitorLoggingAutomation -SettingsFile "c:/directory/settingsFile.yml"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/vm/Set-CmAzIaasUpdateManagement.ps1:20]: Should be Set-CmAzIaasUpdateManagement

File: Cloudmarque.Azure/Public/Deploy/IaaS/vm/Set-CmAzIaasUpdateManagement.ps1 [hunk 17]
```diff
@@ -37,7 +37,7 @@ function Set-CmAzIaasUpdateManagement {
 
 	Get-InvocationInfo -CommandName $MyInvocation.MyCommand.Name
 
-	if ($PSCmdlet.ShouldProcess((Get-CmAzSubscriptionName), "Deploy Core Logging Automation")) {
+	if ($PSCmdlet.ShouldProcess((Get-CmAzSubscriptionName), "Deploy update management")) {
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/vm/Set-CmAzIaasUpdateManagement.ps1:40]: Deploy VM Update Management

File: Cloudmarque.Azure/Public/Deploy/IaaS/vm/Set-CmAzIaasUpdateManagement.ps1 [hunk 18]
```diff
@@ -126,7 +126,11 @@ function Set-CmAzIaasUpdateManagement {
 		}
 
 		Write-Verbose "Deploying Schedule Management..."
+
+		$deploymentName = Get-CmAzResourceName -Resource "Deployment" -Architecture "IaaS" -Region "Global" -Name "Set-CmAzIaasUpdateManagement"
+
 		New-AzResourceGroupDeployment `
+			-Name $deploymentName `
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/vm/Set-CmAzIaasUpdateManagement.ps1:133]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/vpngw/New-CmAzIaasVpnGw.ps1 [hunk 19]
```diff
@@ -171,9 +171,12 @@
 				Set-GlobalServiceValues -GlobalServiceContainer $SettingsObject -ServiceKey "virtualNetworkGateway" -ResourceServiceContainer $vpnGw
 				Set-GlobalServiceValues -GlobalServiceContainer $SettingsObject -ServiceKey "localNetworkGateway" -ResourceServiceContainer $vpnGw
 			}
+			Write-Verbose "Deploying Vpn Gateways..."
+
+			$deploymentName = Get-CmAzResourceName -Resource "Deployment" -Architecture "IaaS" -Region $SettingsObject.location -Name "New-CmAzIaasVpnGw"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/vpngw/New-CmAzIaasVpnGw.ps1:176]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/wvd/New-CmAzIaaSWVD.ps1 [hunk 20]
```diff
@@ -180,8 +180,12 @@ function New-CmAzIaaSWVD {
 				$logAnalyticsID = (Get-CmAzService -Service $SettingsObject.logAnalyticsTag -ThrowIfUnavailable).resourceId
 			}
 
-			Write-Verbose "Deploying WVD infrastructure."
+			Write-Verbose "Deploying WVD infrastructure..."
+
+			$deploymentNameEnv = Get-CmAzResourceName -Resource "Deployment" -Architecture "IaaS" -Region $SettingsObject.azureDeploymentLocation -Name "New-CmAzWVD-Env"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/wvd/New-CmAzIaaSWVD.ps1:185]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/wvd/New-CmAzIaaSWVD.ps1 [hunk 21]
```diff
@@ -230,13 +234,21 @@ function New-CmAzIaaSWVD {
 			}
 
 			Write-Verbose "Deploying WVD hosts."
+
+			$deploymentNameHosts = Get-CmAzResourceName -Resource "Deployment" -Architecture "IaaS" -Region $SettingsObject.azureDeploymentLocation -Name "New-CmAzWVD-Hosts"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/wvd/New-CmAzIaaSWVD.ps1:238]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/IaaS/wvd/New-CmAzIaaSWVD.ps1 [hunk 22]
```diff
@@ -230,13 +234,21 @@ function New-CmAzIaaSWVD {
...
+
 			New-AzDeployment `
+				-Name $deploymentNameHosts `
 				-Location $SettingsObject.azureDeploymentLocation `
 				-TemplateFile "$PSScriptRoot\New-CmAzWVDHosts.json" `
 				-Environments $SettingsObject.wvdEnvironments
 
 			Write-Verbose "Deploying WVD host post setup."
+
+			$deploymentNamePs = Get-CmAzResourceName -Resource "Deployment" -Architecture "IaaS" -Region $SettingsObject.azureDeploymentLocation -Name "New-CmAzWVD-Ps"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/wvd/New-CmAzIaaSWVD.ps1:248]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/PaaS/function/New-CmAzPaasFunction.ps1 [hunk 23]
```diff
@@ -198,21 +198,27 @@ function New-CmAzPaasFunction {
 
                 $region = $SettingsObject.functionAppSolutions.AppServicePlans.Region.count -gt 1 ? $SettingsObject.functionAppSolutions.AppServicePlans[0].Region : $SettingsObject.functionAppSolutions.AppServicePlans.Region
 
-                Write-Verbose "Initiating deployment of functions on app service plans..."
+                Write-Verbose "Deploying functions on app service plans..."
 
-                New-AzDeployment  `
+                $deploymentNameAsp = Get-CmAzResourceName -Resource "Deployment" -Architecture "PaaS" -Region $region -Name "New-CmAzPaasFunction"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/PaaS/function/New-CmAzPaasFunction.ps1:203]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/PaaS/function/New-CmAzPaasFunction.ps1 [hunk 24]
```diff
@@ -198,21 +198,27 @@ function New-CmAzPaasFunction {
...
 
             if ($SettingsObject.functionAppSolutions.ConsumptionPlans) {
 
-                $SettingsObject.functionAppSolutions.ConsumptionPlans.Region.count -gt 1 ? $SettingsObject.functionAppSolutions.ConsumptionPlans[0].Region : $SettingsObject.functionAppSolutions.ConsumptionPlans.Region
+                $region = $SettingsObject.functionAppSolutions.ConsumptionPlans.Region.count -gt 1 ? $SettingsObject.functionAppSolutions.ConsumptionPlans[0].Region : $SettingsObject.functionAppSolutions.ConsumptionPlans.Region
+
+                Write-Verbose "Deploying functions on consumption plans..."
 
-                Write-Verbose "Initiating deployment of functions on consumption plans..."
+                $deploymentNameCon = Get-CmAzResourceName -Resource "Deployment" -Architecture "PaaS" -Region $region -Name "New-CmAzPaasFunction"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/PaaS/function/New-CmAzPaasFunction.ps1:218]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/PaaS/sharedImageGallery/New-CmAzPaasSharedImageGallery.ps1 [hunk 25]
```diff
@@ -131,8 +131,12 @@
 
             }
 
+            Write-Verbose "Deploying Shared Image Gallery..."
+
+            $deploymentName = Get-CmAzResourceName -Resource "Deployment" -Architecture "Paas" -Region $SettingsObject.location -Name "New-CmAzPaasSharedImageGallery"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/PaaS/sharedImageGallery/New-CmAzPaasSharedImageGallery.ps1:136]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/PaaS/sql/New-CmAzPaasSql.ps1 [hunk 26]
```diff
@@ -197,11 +197,16 @@
 
 			$joinedSqlServers = @($servers, $sharedServers)
 
+			Write-Verbose "Deploying Sql Servers and databases..."
+
 			$joinedSqlServers | ForEach-Object {
 
 				if ($_) {
 
+					$deploymentName = Get-CmAzResourceName -Resource "Deployment" -Architecture "PaaS" -Region $SettingsObject.Location -Name "New-CmAzPaasSql"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/PaaS/sql/New-CmAzPaasSql.ps1:206]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/PaaS/web-static/New-CmAzPaasWebStatic.ps1 [hunk 27]
```diff
@@ -97,7 +97,11 @@
 		$storageName = Get-CmAzResourceName -Resource "Storageaccount" -Architecture "IaaS" -Region $SettingsObject.Location -Name $SettingsObject.Name
 
 		Write-Verbose "Deploying cdn: $profileName..."
+
+		$deploymentNameCdn = Get-CmAzResourceName -Resource "Deployment" -Region $SettingsObject.Location -Architecture "PaaS" -Name "New-CmAzPaasWebStatic-cdn"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/PaaS/web-static/New-CmAzPaasWebStatic.ps1:101]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/PaaS/web/New-CmAzPaasWeb.ps1 [hunk 28]
```diff
@@ -244,10 +244,12 @@
 				[System.Collections.ArrayList]$AppServiceDetails = @()
 				$AppServiceDetails += $SettingsObject.WebSolutions.AppServicePlans
 
-				Write-Verbose "Initiating deployment of webapps..."
+				Write-Verbose "Deploying Webapps..."
+
+				$deploymentNameWeb = Get-CmAzResourceName -Resource "Deployment" -Region $region -Architecture "PaaS" -Name "New-CmAzPaasWeb-WebApp"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/PaaS/web/New-CmAzPaasWeb.ps1:249]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/PaaS/web/New-CmAzPaasWeb.ps1 [hunk 29]
```diff
@@ -285,10 +287,12 @@
 				[System.Collections.ArrayList]$ApiManagementServices = @()
 				$ApiManagementServices += $SettingsObject.WebSolutions.ApiManagementServices
 
-				Write-Verbose "Initiating deployment of api management services..."
+				Write-Verbose "Deploying api management services..."
 
-				New-AzDeployment  `
-					-Name "CmAzApiManagementServices_Parent" `
+				$deploymentNameApim = Get-CmAzResourceName -Resource "Deployment" -Region $region -Architecture "PaaS" -Name "New-CmAzPaasWeb-ApiMs"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/PaaS/web/New-CmAzPaasWeb.ps1:292]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/PaaS/web/New-CmAzPaasWeb.ps1 [hunk 30]
```diff
@@ -486,8 +490,12 @@
 					}
 				}
 
+				Write-Verbose "Deploying Frontdoor..."
+
+				$deploymentNameFd =  Get-CmAzResourceName -Resource "Deployment" -Region "Global" -Architecture "PaaS" -Name "New-CmAzPaasWeb-Fd"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/PaaS/web/New-CmAzPaasWeb.ps1:495]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Public/Deploy/Security/security-centre/Set-CmAzSecurityCentre.ps1 [hunk 31]
```diff
@@ -56,7 +56,11 @@ function Set-CmAzSecurityCentre {
             Get-AzSecurityContact | Remove-AzSecurityContact
 
             Write-Verbose "Deploying security centre settings..."
+
+            $deploymentName = Get-CmAzResourceName -Resource "Deployment" -Region $SettingsObject.Location -Architecture "SaaS" -Name "Set-CmAzSecurityCentre"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/Security/security-centre/Set-CmAzSecurityCentre.ps1:60]: Can set this directly on the deployment cmdlet

File: Cloudmarque.Azure/Resources/Project/_names/generators.yml [hunk 32]
```diff
@@ -8,6 +8,17 @@ formats:
       - source: environment
       - source: region
       - source: hash
+  - name: deployment
+    separator: "-"
+    case: lower
+    components:
+      - source: name
+      - source: packageVersion
```
Comment (tm-luke) [inline Cloudmarque.Azure/Resources/Project/_names/generators.yml:16]: Swap environment and package version

**Review by tm-luke — CHANGES_REQUESTED** [review]
Can you perform test deployments in multiple locations please.

File: Cloudmarque.Azure/Public/Deploy/Core/automation/New-CmAzCoreAutomation.ps1 [hunk 33]
```diff
@@ -109,16 +109,22 @@
...
 				-ResourceGroupName $nameResourceGroup `
 				-TemplateFile "$PSScriptRoot\New-CmAzCoreAutomation.json" `
 				-AccountName $automationAccountName `
 				-Location $SettingsObject.location `
 				-AutomationService $SettingsObject.service.publish.automation `
 				-Force > $Null
 
-			# Create Automation account
+			# Create Linked Services
+			$deploymentName = Get-CmAzResourceName -Resource "Deployment" -Architecture "Core" -Region $SettingsObject.location -Name "New-CmAzCoreAutomation.ls"
```
[the remark this answers is not in the document]
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/Deploy/Core/automation/New-CmAzCoreAutomation.ps1:124]: no it is getting too long and we are loosing timestamp

**Review by tm-luke — APPROVED** [review]

**Pull request merged**